mono/mini/exceptions-amd64.c

```diff
@@ -926,8 +926,7 @@ mono_arch_handle_altstack_exception (void *sigctx, MONO_SIG_HANDLER_INFO_TYPE *s
 	 * requires allocation on the stack, as this wouldn't be encoded in unwind
 	 * information for the caller frame.
 	 */
-	sp = (gpointer *)UCONTEXT_REG_RSP (sigctx);
-	g_assertf (((unsigned long) sp & 15) == 0, "sp: %p\n", sp);
+	sp = (gpointer *)(UCONTEXT_REG_RSP (sigctx) & ~15);
```

Comment:
Can we use some “align down” macro or function?
This code is subtle.
It depends on the left being signed, or unsigned and signed combination being signed.
Can you maybe cast the 15 to gsize or gssize?

Reply:
good point, updated the PR.